### ISSUE TITLE
Consider improved Dockerfile generation

### ISSUE THREAD

**Issue by macsux** [dotnet/tye#348]
I suggest integrating what NMica Dockerfile generation does to improve the efficiency of layering. 
https://github.com/NMica/NMica
Here's an in-depth blog post on what it does
http://stakhov.pro/building-efficient-net-docker-images/

**Comment by rynowak** [comment]
Sorry for the delay in getting back to you. I think this is definitely a good idea for when we're publishing. 

So if I'm understanding how this works correctly this package:

- reads the `project.assets.json`, and then categorizes the assets based on where they came from 
- then copies them into separate folders during publish
- the places those into separate docker layers

Is that correct?

**Comment by macsux** [comment]
Yes

**Comment by staff0rd** [comment]
is there some way already to export locally the `Dockerfile` that tye will build?

**Comment by jkotalik** [comment]
[`tye build`](https://github.com/dotnet/tye/blob/master/docs/reference/commandline/tye-build.md) should be what you are looking for.

**Comment by rynowak** [comment]
I don't think so. We don't persist the file to to disk in a way the user can see.

**Comment by jkotalik** [comment]
Ah, build only creates the image without pushing it. I thought it would create the file that would be used for images. We could consider updating the generate command to support it.
